File: python/517_超级洗衣机_hard.py
class Solution:
    def findMinMoves(self, machines: List[int]) -> int:
        n, total = len(machines), sum(machines)
        if total % n:
            return -1
        avg = total // n
        ans, sub = 0, 0
        for num in machines:
            num -= avg
            sub += num
            ans = max(ans, abs(sub), num)
        return ans

File: cpp/517_超级洗衣机_hard.cpp
class Solution {
public:
    int findMinMoves(vector<int>& machines) {
        int n = machines.size();
        int all = accumulate(machines.begin(), machines.end(), 0);
        if (all % n) return -1;
        int avg = all / n;
        int ans = 0, sub = 0;
        for (auto num : machines) {
            num -= avg;
            sub += num;
            ans = max(ans, abs(sub));
            ans = max(ans, num);
        }
        return ans;
    }
};